test/unit/JsonService.test.ts

```diff
@@ -148,5 +182,314 @@ describe("JsonService", () => {
             // assert
             expect(result).toEqual(json);
         });
+
+        it("should work with custom jwtHandler", async () => {
+            // arrange
+            const jwtHandler = jest.fn();
+            subject = new JsonService([], jwtHandler);
+            const text = "text";
+            fetchMock.mockResolvedValue({
+                status: 200,
+                headers: new Headers({
+                    "Content-Type": "application/jwt"
+                }),
+                text: () => Promise.resolve(text)
+            });
+
+            // act
+            await subject.getJson("http://test");
+
+            // assert
+            expect(jwtHandler).toBeCalledWith(text);
+        });
+    });
+
+    describe("postForm", () => {
+        it("should return a promise", async () => {
+            // act
+            const p = subject.postForm("http://test", { "a": "b" });
+
+            // assert
+            expect(p).toBeInstanceOf(Promise);
+            // eslint-disable-next-line no-empty
+            try { await p; } catch {}
+        });
+
+        it("should reject promise when no url is passed", async () => {
+            try {
+                // act
+                await subject.postForm("", { "a": "b" });
+            } catch (err) {
+                // assert
+                expect(err).toBeInstanceOf(Error);
+                expect((err as Error).message).toContain("url");
+            }
+        });
+
+        it("should make POST request to url", async () => {
+            try {
+                // act
+                await subject.postForm("http://test", { "a": "b" });
+            } catch {
```

Comment:
It seems like a lot of these `try/catch` patterns that only test if the request was attempted should actually be `try/finally`s.